[rebar/rebar3] rebar3 update to 3.9.0, deprecate rebar

predominant opened the pull request:
This PR:

* Updates rebar3 to 3.9.0
* Deprecates rebar

Following merge, according to [RFC7](https://github.com/habitat-sh/core-plans-rfcs/blob/master/_RFCs/0007-deprecating-packages.md) we need to:

* Mark `core/rebar` as private
* Disconnect `core/rebar` from builder

### Testing

```
hab studio enter
./rebar3/tests/test.sh
```

### Sample output

```
 ✓ Version matches
 ✓ Help command

2 tests, 0 failures
```

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

bdangit commented on rebar/README.md:
```diff
@@ -1,4 +1,6 @@
-# rebar
+# rebar (deprecated)
+
+**Deprecation Notice**: This package has been deprecated by the upstream developer. Please consider migration to use the `core/rebar3` plan.
```
do we want to require adding the following lines into deprecated plans?

```bash
# Hint for rebuild scripts. Not a formal part of plan-build.
pkg_deprecated="true"
```

bdangit commented on rebar3/plan.sh:
```diff
@@ -24,7 +29,9 @@ do_build() {
   cp -R "_build/default/"* "${pkg_prefix}"
+  cp -R "_build/prod/bin" "${pkg_prefix}"
   fix_interpreter "${pkg_prefix}/bin/"* core/busybox-static bin/env
+  attach
```
doh... you'll need to remove this line.

smacfarlane commented:
@predominant gentle ping on the requested changes

predominant commented:
Removed attach statement, squashed commits.

Regarding adding `pkg_deprecated`, I think this would be a good candidate for RFC7 modification / discussion.

smacfarlane commented:
> Regarding adding `pkg_deprecated`, I think this would be a good candidate for RFC7 modification / discussion.

@predominant  agreed, and I'll try and put something together this week.  Would you be opposed to adding that in the meantime though?  

It boils down to needing a way to prune deprecated plans when operating in "rebuild the world" scenarios when builder is not involved, such as a base plans refresh.

predominant commented:
Yeah, no problem adding it now @smacfarlane @bdangit ! Done and Done!
